mesh/mesh.cpp

```diff
@@ -3776,19 +3812,21 @@ Mesh::Mesh(Mesh *orig_mesh, int ref_factor, int ref_type)
 
    for (int el = 0; el < orig_mesh->GetNE(); el++)
    {
-      Geometry::Type geom = orig_mesh->GetElementBaseGeometry(el);
+      int ref = ref_factors[el];
+
+      Geometry::Type geom = orig_mesh->GetElementGeometry(el);
       int attrib = orig_mesh->GetAttribute(el);
       int nvert = Geometry::NumVerts[geom];
-      RefinedGeometry &RG = *GlobGeometryRefiner.Refine(geom, ref_factor);
+      RefinedGeometry &RG = *GlobGeometryRefiner.Refine(geom, ref);
 
       max_nv = std::max(max_nv, nvert);
       rfes.GetElementDofs(el, rdofs);
       MFEM_ASSERT(rdofs.Size() == RG.RefPts.Size(), "");
       const FiniteElement *rfe = rfes.GetFE(el);
       orig_mesh->GetElementTransformation(el)->Transform(rfe->GetNodes(),
                                                          phys_pts);
-      const int *c2h_map = rfec.GetDofMap(geom);
-      const int *vertex_map = vertex_fec.GetDofMap(geom);
+      const int *c2h_map = rfec.GetDofMap(geom, ref);
+      const int *vertex_map = vertex_fec.GetDofMap(geom, 1);
```

Comment:
magic number.

Reply:
I think this should work too, would need to double-check:
```suggestion
      const int *vertex_map = vertex_fec.GetDofMap(geom);
```